drivers/sx126x/include/sx126x_internal.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sx126x_internal SX1261/2/8 and LLCC68 internal functions
+ * @ingroup     drivers_sx126x
+ * @brief       Internal functions for the SX1261/2/8 and LLCC68
+ *
+ * @{
+ *
+ * @file
+ *
+ * @author      José I. Alamos <jose.alamos@haw-hamburg.de>
+ */
+#ifndef SX126X_INTERNAL_H
+#define SX126X_INTERNAL_H
+
+#include "sx126x.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief     Check the variant of sx126x
+ *
+ * @warning   This only works if @ref SX126X_SINGLE is not set! This means,
+ *            more than one variant is present. If @ref SX126X_SINGLE is set,
+ *            this functions asserts.
+ *
+ * @param[in] dev                       Device descriptor.
+ * @param[in] type                      Variant of the driver to compare to.
+ *
+ * @return    true if @p type matches the variant type
+ * @return    false otherwise
+ */
+static inline bool sx126x_check_type(sx126x_t *dev, sx126x_type_t type)
+{
+#if SX126X_SINGLE
+    (void) dev;
+    (void) type;
+    assert(false);
+    return false;
```

Comment:
Here and for the rest?
```suggestion
assert(dev);
#if SX126X_SINGLE
    (void) type;
    assert(false);
    return false;
```

Reply:
addressed! (I removed this function though because the `type` member is not guarded anymore)